### PR TITLE
feat: add azure support

### PR DESCRIPTION
**Pull request by leo220yuyaodog**

### CODE REVIEW

File: azureblob/azure.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
+package azureblob
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/Azure/azure-storage-blob-go/azblob"
```
Comment (nomeguy) [inline azureblob/azure.go]: sort

File: azureblob/azure.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
+package azureblob
```
Comment (nomeguy) [inline azureblob/azure.go:15]: header

File: azureblob/azure.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+)
+
+// Client azure blob storage
+type Client struct {
+	Config       *Config
+	containerURL *azblob.ContainerURL
+}
+
+type Config struct {
+	AccessID  string //Account Name
```
Comment (nomeguy) [inline azureblob/azure.go]: Id

File: azureblob/azure.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+const blobFormatString = `https://%s.blob.core.windows.net`
+
+var (
+	ctx = context.Background() // This uses a never-expiring context.
+)
+
+func New(config *Config) *Client {
+
```
Comment (nomeguy) [inline azureblob/azure.go]: no line

File: azureblob/azure.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+	var client = &Client{Config: config}
+
+	serviceURL, _ := GetBlobService(config)
+	client.containerURL = containerURL(serviceURL, config)
+	return client
+}
+
+func GetBlobService(config *Config) (azblob.ServiceURL, error) {
+
```
Comment (nomeguy) [inline azureblob/azure.go]: no line

File: azureblob/azure.go [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+	// From the Azure portal, get your Storage account blob service URL endpoint.
+	// The URL typically looks like this:
+	u, _ := url.Parse(fmt.Sprintf(blobFormatString, config.AccessID))
+
+	// Create an ServiceURL object that wraps the service URL and a request pipeline.
+	return azblob.NewServiceURL(*u, p), nil
+}
+
+func containerURL(serviceURL azblob.ServiceURL, config *Config) *azblob.ContainerURL {
```
Comment (nomeguy) [inline azureblob/azure.go]: Url

File: azureblob/azure_test.go [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package azureblob
```
Comment (nomeguy) [inline azureblob/azure_test.go:15]: header

File: azureblob/azure_test.go [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/qor/oss/tests"
+)
+
+func TestClient_Get(t *testing.T) {
```
Comment (nomeguy) [inline azureblob/azure_test.go]: Remove this

File: azureblob/azure_test.go [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+}
+
+var client *Client
+
+func init() {
+	client = New(&Config{
+		AccessID:  "AKIDToxukQWBG8nGXcBN8i662nOo12sc5Wjl",
+		AccessKey: "40jNrBf5mLiuuiU8HH7lDTXP5at00sbA",
```
Comment (nomeguy) [inline azureblob/azure_test.go]: Don't store keys here

File: azureblob/azure_test.go [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+		Bucket:    "tets-1252882253",
+		Region:    "ap-shanghai",
+		Endpoint:  "localhost:8080",
+	})
+}
+
+func TestClient_Put(t *testing.T) {
```
Comment (nomeguy) [inline azureblob/azure_test.go]: No _

**Review by nomeguy — CHANGES_REQUESTED** [review]

**Comment by leo220yuyaodog** [comment]
@nomeguy I resolve the mistakem plz review.

**Pull request closed without merging**